syntax = "proto3";

service ManticoreUI {
    rpc Terminate(ManticoreInstance) returns (TerminateResponse) {}
    rpc Start(CLIArguments) returns (ManticoreInstance) {}
    rpc TargetAddress(AddressRequest) returns (TargetResponse) {}
}

message ManticoreInstance {
    string uuid = 2;
}

message TerminateResponse { 
    bool success = 3;
}

message CLIArguments {
    string program_path = 4;
    repeated string binary_args = 5;
    repeated string envp = 6;
    repeated string symbolic_files = 7;
    string concrete_start = 8;
    string stdin_size = 9;
    map<string, string> additional_mcore_args = 10;
}

message AddressRequest {
   
    enum TargetType {
        FIND = 0;
        AVOID = 1;
        CLEAR = 2;
    }
    
    ManticoreInstance mcore_instance = 11;
    uint64 address = 12;
    TargetType type = 13;
}

message TargetResponse {
    bool success = 14;
}
